aws-rds-dbinstance/src/main/java/software/amazon/rds/dbinstance/CreateHandler.java

```diff
@@ -210,10 +216,29 @@ private String fetchEngine(final ProxyClient<RdsClient> client, final ResourceMo
         }
 
         if (ResourceModelHelper.isDBInstanceReadReplica(model)) {
-            return fetchDBInstance(client, model.getSourceDBInstanceIdentifier()).engine();
+            final String sourceDBInstanceArn = model.getSourceDBInstanceIdentifier();
+            final String sourceDBInstance = ResourceModelHelper.isValidArn(sourceDBInstanceArn) ?
```

Comment:
I think we could be a bit more defensive here. If a customer passes on a malformed ARN we are treating it just like the customer is passing an actual DBInstance ID. The concern is that in the future we could simply look at the variable name and make wrong assumptions. Perhaps naming it as sourceDBInstanceIdOrArn would make it explicit for future developers to avoid assumption traps.

Reply:
Done.